### PR TITLE
* [ios] Refactor transform and transition 

### PR DESCRIPTION
**Pull request by cxfeng1**
### Refactor transform and transition
  * UIView animation does not support rotation animation whose degree > 180，so we use CAAnimation to hack it.
  * Using UIView animation along with CAAnimation has a lot of problems when developers use rotate/scale/translate together
  * We refactor all transition properties to use CAAnimation to ensure consistency
  * We also separate transform string parsing with runtime parsing to ensure performance
  * tested demos:
     * all the demos in playground and tests 
     * http://dotwe.org/weex/7d8e99a7d5dc8ee22e2437d6d8513fec
     * http://dotwe.org/weex/cad4d4cd051ab3e565f3cd42bbcac73b
     * http://dotwe.org/weex/8cfadb65d744fd400728097ef7da3bcc
 * related issue: #1596 #2070 #2182 #2226

### CODE REVIEW

**Comment by cxfeng1** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><strong><a href="https://github.com/cxfeng1/weex/blob/2d51ef7d57873a0d37db46e0a25968ba42513441/ios/sdk/WeexSDK/Sources/Module/WXAnimationModule.m#L48">ios/sdk/WeexSDK/Sources/Module/WXAnimationModule.m#L48</a></strong>: cannot find protocol declaration for ‘CAAnimationDelegate’; did you mean ‘UIApplicationDelegate’?  <br /><code>
@interface WXAnimationDelegate : NSObject &lt;CAAnimationDelegate&gt;
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: ios/sdk/WeexSDK/Sources/Module/WXAnimationModule.m [hunk 1]
```diff
@@ -37,59 +112,106 @@ - (void)transition:(NSString *)nodeRef args:(NSDictionary *)args callback:(WXMod
...
+            newBounds.size = CGSizeMake(newBounds.size.width, [WXConvert WXPixelType:value scaleFactor:self.weexInstance.pixelScaleFactor]);
+            info.toValue = [NSValue valueWithCGRect:newBounds];
+            [infos addObject:info];
         }
     }
-    
+
+    return infos;
+}
+
```
Comment (kfeagle) [inline ios/sdk/WeexSDK/Sources/Module/WXAnimationModule.m]: 可以去掉

File: ios/sdk/WeexSDK/Sources/Module/WXAnimationModule.m [hunk 2]
```diff
@@ -8,9 +8,84 @@
...
+        WXTransform *transform = _animationInfo.target->_transform;
+        [transform applyTransformForView:_animationInfo.target.view];
+    } else if ([_animationInfo.propertyName isEqualToString:@"backgroundColor"]) {
+        _animationInfo.target.view.layer.backgroundColor = (__bridge CGColorRef _Nullable)(_animationInfo.toValue);
+    } else if ([_animationInfo.propertyName isEqualToString:@"opacity"]) {
+        _animationInfo.target.view.layer.opacity = [_animationInfo.toValue floatValue];
+    }
+}
+
+- (void)animationDidStop:(CAAnimation *)anim finished:(BOOL)flag
```
Comment (kfeagle) [inline ios/sdk/WeexSDK/Sources/Module/WXAnimationModule.m]: flag换成finished感觉更好

File: ios/sdk/WeexSDK/Sources/Module/WXAnimationModule.m [hunk 3]
```diff
@@ -37,59 +112,106 @@ - (void)transition:(NSString *)nodeRef args:(NSDictionary *)args callback:(WXMod
...
                  Rotate >= 180 degree not working on UIView block animation, have not found any more elegant solution than using CAAnimation
                  See http://stackoverflow.com/questions/9844925/uiview-infinite-360-degree-rotation-animation
                  **/
-                isUsingCAAnimation = YES;
+                newInfo.fromValue = @(oldTransform.rotateAngle);
+                newInfo.toValue = [NSNumber numberWithDouble:wxTransform.rotateAngle];
+                [infos addObject:newInfo];
+            }
+            
+            if (wxTransform.scaleX != oldTransform.scaleX) {
```
Comment (kfeagle) [inline ios/sdk/WeexSDK/Sources/Module/WXAnimationModule.m]: 这种比较会有隐藏风险，出现问题排查难，我等下为weex写一个比较函数

**Pull request merged**